src/controllers/CheckYourAppealController.ts

```diff
@@ -1,24 +1,36 @@
 import { controller, BaseHttpController, httpGet } from 'inversify-express-utils';
 import { SUBMISSION_SUMMARY_PAGE_URI } from '../utils/Paths';
+import { SessionKey } from 'ch-node-session-handler/lib/session/keys/SessionKey';
+import { SignInInfoKeys } from 'ch-node-session-handler/lib/session/keys/SignInInfoKeys';
+import { ISignInInfo } from 'ch-node-session-handler/lib/session/model/SessionInterfaces';
+import { Request } from 'express';
+import { AuthMiddleware } from '../middleware/AuthMiddleware';
+import { SessionMiddleware } from 'ch-node-session-handler';
 
-@controller(SUBMISSION_SUMMARY_PAGE_URI)
+@controller(SUBMISSION_SUMMARY_PAGE_URI, SessionMiddleware, AuthMiddleware)
 export class CheckYourAppealController extends BaseHttpController {
 
     @httpGet('')
-    public renderView(): void {
+    public renderView(req: Request): void {
 
-        const session = this.httpContext.request.session;
-        let userProfile = '';
-        let reasons = '';
-        let penaltyIdentifier = '';
+        const userProfile = req.session
+            .chain(_ => _.getValue<ISignInInfo>(SessionKey.SignInInfo))
+            .map(info => info[SignInInfoKeys.UserProfile]);
 
+        const appealsData = req.session
+            .chain(_ => _.getExtraData())
+            .map(data => {
+                return {
+                    penaltyIdentifier: data.appeals.penaltyIdentifier,
+                    reasons: data.appeals.reasons
+                };
+            });
 
-        if(session){
-            penaltyIdentifier = session.getExtraData('appeals').penaltyIdentifier;
-            reasons = session.getExtraData('appeals').reasons;
-            userProfile = session.getValue('signin_info').user_profile;
-        }
+        userProfile
```

Comment:
Mapping user profile and then getting appeals details does not make sense. Make just simple call to `this.httpContext.response.render` function passing both userProfile and appealsData. That mapping is absolutely necessary.